packages/client/src/components/sidebar/auth-state.js

```diff
@@ -0,0 +1,283 @@
+
+        this.emitter.emit('updatesession', session.id)
+        this.machine.emit('login')
+      } catch (error) {
+        console.error('sidebar/auth-state: error fetching session', error)
+
+        api.clearSession()
+
+        this.emitter.emit('updatesession', null)
+        this.machine.emit('logout')
+      }
+    })
+
+    this._cache = {}
+    this.update() // set cache
+  }
+
+  createElement () {
+    css`
+      #sidebar .auth-state {
+        display: flex;
+        align-items: center;
+        padding: 0 16px;
+
+        font-size: 14px;
+        font-weight: normal;
+        color: var(--gray-300);
+
+        & > .text {
+          margin-right: auto;
+        }
+
+        & > button {
+          padding: 6px 8px;
+
+          background: var(--blue-a2);
+          color: var(--blue);
+
+          cursor: pointer;
+          font-size: 12px;
+
+          border-radius: 4px;
+          border: 0;
+
+          &:hover {
+            background: var(--blue-a3);
+          }
+
+          &.minor {
+            margin-left: 8px;
+
+            background: var(--gray-700);
+            color: var(--gray-300);
+
+            &:hover {
+              background: var(--gray-700);
+              color: var(--gray-100);
+            }
+          }
+        }
+      }
+    `
+
+    if (this.machine.state === 'unloaded') {
+      return html`
+        <section class='auth-state'>
+          <!-- auth state -->
+        </section>
+      `
+    }
+
+    if (this.machine.state === 'fetching') {
+      return html`
+        <section class='auth-state loading'>
+          <div class='text'>Logging in...</div>
+        </section>
+      `
+    }
+
+    if (this.machine.state === 'loggedin') {
+      css`
+        #sidebar .auth-state.logged-in {
+          & .username {
+            color: var(--blue);
+            cursor: pointer;
+            text-decoration: underline;
+            font-weight: bold;
+
+            margin-left: 5px;
+          }
+        }
+      `
+
+      return html`
+        <section class='auth-state logged-in'>
+          <div class='text'>
+            Logged in as
+            <a class='username' onclick=${() => this.openAccountSettings()}>${api.session.user.username}</a>
+          </div>
+        </section>
+      `
+    }
+
+    if (this.machine.state === 'loggedout') {
+      return html`
+        <section class='auth-state logged-out'>
+          <div class='text'>Logged out</div>
+
+          <button onclick=${() => this.displayRegisterModal()}>Register</button>
+          <button class='minor' onclick=${() => this.displayLoginModal()}>Login</button>
+        </section>
+      `
+    }
+
+    if (this.machine.state === 'needsauth') {
+      css`
+        #sidebar .auth-state.needs-auth {
+          color: var(--blue);
+          background: var(--gray-200);
+        }
+      `
+
+      return html`
+        <section class='auth-state needs-auth'>
+          <span class='msg'>needs auth</span>
```

Comment:
Is this temporary? I'm pretty sure this should display the server authorization message.

Reply:
Oh, yep!